Updating Config V2 (#355)
configcat/docs: website/docs/advanced/config-v2-migration-guide.md

Read more about the new features in the [Config V2 Overview](/advanced/config-v2).

## A few things to consider before migration

- Migration from Config V1 means that a V2 config will be copied from the V1 config. The V1 config will remain unchanged and accessible. The V2 version will have a new SDK key for each environment.
- Config V2 and V1 are completely separate. They don't affect each other in any way. You can use them side by side.
- Every newly created config will be a V2 config by default.
- There is no automatic sync between the V1 and V2 configs. If you want to keep them in sync, you have to manually update the V2 config when you make changes to the V1 config and vice versa.
- There is no pressure to migrate. Although we have plans to phase out V1 eventually, you can stay on it for a long time.
- Once the migration process is started, it's recommended to migrate your V1 configs to V2 as quickly as possible to avoid confusion. 

## Migrating from Config V1 to Config V2

> To migrate from Config V1 to Config V2, you have to copy your V1 config to a new V2 one and update your applications to use the new config. The migration process is the following:

Reviewer: Ez nem igazán pontos megfogalmazás, még mindig áll, amit Lali írt: "A you have to create rész szerintem nem fedi így a valóságot, mert így én úgy tudom érteni, hogy csinálni kell egy új configot, de az egy másik flow."

```suggestion
The migration process starts with copying your V1 config to a new V2 one and updating your applications to use the new config. The complete process is the following:
```

## What is Config V2?

> Config V2 is the next generation of ConfigCat. It comes with a new dashboard, API, SDKs, and features. The V1 Dashboard, API, and SDKs are not compatible with V2. However the V2 SDKs are backwards compatible with V1.

Author: Nekem ez mar sok. Felek hogy elveszik az ollvaso a reszletekben

> Config V2 is the next generation of ConfigCat. It comes with a new dashboard, API, SDKs, and features. The V1 Dashboard, API, and SDKs are not compatible with V2. However the V2 SDKs are backwards compatible with V1.

Reviewer: Talán egy picit szájbarágós, de én nem bánnám, ha az elején tiszta vizet öntenénk a pohárba.

> Config V2 is the next generation of ConfigCat. It comes with a new dashboard, API, SDKs, and features. The V1 Dashboard, API, and SDKs are not compatible with V2. However the V2 SDKs are backwards compatible with V1.

Reviewer: Esetleg ha ezt kivennénk belőle: "Also, the new SDK versions that support Config V2 are backwards compatible with V1, that is, they work with both just fine." ? Talán ez a közbevetés töri meg a gondolatmenetet egy kissé.

> Config V2 is the next generation of ConfigCat. It comes with a new dashboard, API, SDKs, and features. The V1 Dashboard, API, and SDKs are not compatible with V2. However the V2 SDKs are backwards compatible with V1.

Author: Ha ragszkodsz hozza , akkor talan inkabb egy "hogyan mukodik " szekcioba lenne erdemes ezeket rakni

Read more about the new features in the [Config V2 Overview](advanced/config-v2).

## A few things to consider before migration

- Config V2 and V1 are completely separate. They don't affect each other in any way. You can use them side by side.
- After Config V2 launches, every newly created config will be a V2 config by default.

> - You can migrate your V1 configs to V2. That will mean that a V1 and a V2 version of your config will be available at the same time. The V2 version will have a new SDK key for each environment.

Reviewer: Itt is lehetne még egy kicsit pontosítani a magyarázatot:

```suggestion
- You can migrate your V1 configs to V2. That means that a V2-enabled copy of your V1 config will be created, that is, you will end up with a V1 and V2 version of the migrated config, which will exist side by side. The V2 version will have a new SDK key for each environment.
```

- There is no automatic sync between the V1 and V2 configs. If you want to keep them in sync, you have to manually update the V2 config when you make changes to the V1 config and vice versa.
- Considering the above, it's recommended to migrate your V1 configs to V2 as quickly as possible to avoid confusion. However, there is no pressure to migrate. Although we have plans to phase out V1 eventually, you can stay on it for a long time.

> - There is no pressure to migrate. You can stay on V1 for a long time.

Reviewer 2: Ez a rész ismétlődik az előző pont átírása után.

## Migrating from Config V1 to Config V2

> To migrate from Config V1 to Config V2, you have to create a new V2 config and update your applications to use the new config. The migration process is the following:

Reviewer 2: A you have to create rész szerintem nem fedi így a valóságot, mert így én úgy tudom érteni, hogy csinálni kell egy új configot, de az egy másik flow.

### Step 1: Create the V2 config

1. Open the ConfigCat Dashboard and click on the `Start migration` button on top of a V1 config.
2. Click `Create V2 config` to create a new config with the same settings as the V1 config.

It's important to note that the V2 config will be created with the same settings as the V1 config and the V1 config will still be accessible and unchanged.

### Step 2: Upgrade the ConfigCat SDK version

> In your application, replace the old ConfigCat SDK Key with the new one (i.e. with the one that belongs to the V2 version of the config) in every environment. The new key can be found on the ConfigCat Dashboard on the V2 config's page top right corner.

Reviewer: Itt összekeveredtek egy kicsit a dolgok, ez a Step 3-ba jön.
